eos-paygd/main.c

```diff
@@ -186,15 +245,41 @@ main (int   argc,
   const gchar *sd_socket_env = NULL;
   g_autofree char *sd_socket_dir = NULL;
   g_autofree char *sd_socket_name = NULL;
+  GOptionContext *context;
 
+  context = g_option_context_new ("- Pay As You Go enforcement daemon");
+  g_option_context_add_main_entries (context, opts, GETTEXT_PACKAGE);
+  if (!g_option_context_parse (context, &argc, &argv, &error))
+    {
+      g_printerr ("%s: %s\n", argv[0], error->message);
+      return GOPTION_FAILURE_EXIT_CODE;
+    }
+  if (print_level)
+    {
+      printf ("%c", EPG_SECURITY_LEVEL);
```

Comment:
Really appreciate the speedy review on this one.  Every other comment is indisputable, and I'll address those with a revised patch shortly.

This one I'd like to push back against - no human should ever be looking at this value, the only reason there's help text for options at all is because glib provided it when I used the glib standard functions. (my preference would've been to use getopt and not provide --help at all, but I felt that would've violated project coding style)

Currently this value is a raw char, and my intended way to use it is:
eospaygd -s > /sys/firmware/efi/efivars/EOSPAYG_securitylevel-guid

The read code doesn't change at all if we have a newline there, but I don't think there's any benefit to trying to make this more human readable - we're still in libexec (or /usr/lib/x86-64-whatever) and not a standard bin dir, so I don't think there's any expectation that a user can interact with this.

Reply:
Re: stdout being line buffered, the exit() immediately following the printf will implicitly flush the buffers

Reply:
Okay, sounds good